src/perception/road_signs_classifier/road_signs_classifier/road_signs_classifier.py

```diff
@@ -0,0 +1,76 @@
+"""
+Package: road_signs_classifier
+File: road_signs_classifier.py
+Author: Pranav Boyapati
+"""
+
+import rclpy
+import base64
+import os
+from rclpy.node import Node
+from sensor_msgs.msg import Image
+from cv_bridge import CvBridge
+from navigator_msgs.msg import RoadSigns
+from navigator_msgs.msg import RoadSignsDetection
+import cv2
+from inference_sdk import InferenceHTTPClient
+
+
+class road_signs_classifier(Node):
+    def __init__(self):
+        super().__init__('road_signs_classifier')
+
+        #instantiate model client
+        self.CLIENT = InferenceHTTPClient(api_url="http://localhost:6060", api_key="BmqYjCBXZD1iPIyq09sG")
+
+        #create subscriptions
+        self.camera_sub = self.create_subscription(Image, '/cameras/camera0', self.image_callback, 10)
+
+        #create variables to store subscription info
+        self.bridge = CvBridge()
+
+        #create publisher
+        self.road_signs_publisher = self.create_publisher(RoadSignsDetection, '/road_signs/detections', 10)
+
+
+    #callbacks for subscriptions
+    def image_callback(self, msg : Image):
+        cv_image = self.bridge.imgmsg_to_cv2(msg, desired_encoding="passthrough") 
+        cv2.imwrite("image.jpeg", cv_image) 
+        self.classify_sign()
+
+    
+    #main control function
+    def classify_sign(self):
+        #gets prediction
+        if (os.path.exists("image.jpeg")):
+            with open("image.jpeg", "rb") as img_file:
+                img_base64 = base64.b64encode(img_file.read()).decode('utf-8')
+            result = self.CLIENT.infer(img_base64, model_id="us-road-signs/71")
+            print(result)
+
+        #create message and publish
+        if (len(result["predictions"]) > 0):
+
+            road_signs_detection_msg = RoadSignsDetection()
+            road_signs_detection_msg.header.stamp = self.get_clock().now().to_msg()
+            road_signs_detection_msg.header.frame_id = "camera_frame"
+            
+            road_sign = RoadSigns()
+            road_sign.x = result["predictions"][0]['x']
+            road_sign.y = result["predictions"][0]['y']
+            road_sign.width = result["predictions"][0]['width']
+            road_sign.height = result["predictions"][0]['height']
+            road_sign.label = result["predictions"][0]['class']
+            road_sign.confidence = result["predictions"][0]['confidence']
+
+            road_signs_detection_msg.road_signs.append(road_sign)
+
+            self.road_signs_publisher.publish(road_signs_detection_msg)
+
+
+if __name__ == '__main__':
+    rclpy.init(args=None)
+    road_signs_classification = road_signs_classifier()
+    rclpy.spin(road_signs_classification)
+    rclpy.shutdown()
```

Comment:
Can we add a`main` function? For future reference, `ros2 run road_sign_classifier road_sign_classifier` looks for a main function. This is in case someone wants to test it in isolation isolated in the future.
See this for reference:
https://github.com/Nova-UTD/navigator/blob/ed79ad2a60f97737e03953382b7470e1a06780b8/src/planning/rtp/rtp/rtp_node.py#L433-L448